quickjs.c

```diff
@@ -32777,8 +32777,10 @@ static JSValue js_create_function(JSContext *ctx, JSFunctionDef *fd)
             }
         } else {
             b->vardefs = (void *)((uint8_t*)b + vardefs_offset);
-            memcpy(b->vardefs, fd->args, fd->arg_count * sizeof(fd->args[0]));
-            memcpy(b->vardefs + fd->arg_count, fd->vars, fd->var_count * sizeof(fd->vars[0]));
+            if (fd->vars && fd->arg_count > 0) {
+                memcpy(b->vardefs, fd->args, fd->arg_count * sizeof(fd->args[0]));
+                memcpy(b->vardefs + fd->arg_count, fd->vars, fd->var_count * sizeof(fd->vars[0]));
```

Comment:
It looks like this will cause a regression: if only one of the two is empty, the other one won't be copied. Shouldn't we have two checks here?